pysb/simulator/cupsoda.py

```diff
@@ -474,12 +474,13 @@ def _get_cmatrix(self):
             for j in range(self._len_rxns):
                 rate = 1.0
                 for r in rate_args[j]:
-                    x = str(r)
-                    if x in par_names:
-                        rate *= par_vals[i][par_dict[x]]
+                    if isinstance(r, pysb.Parameter):
+                        rate *= par_vals[i][par_names.index(r.name)]
```

Comment:
It didn't cause a problem, but I was simplifying the code when making some other changes and didn't revert this change back. I would imagine the performance difference is negligible in any case.